src/main/java/dev/morling/onebrc/CalculateAverage_JaimePolidura.java

```diff
@@ -0,0 +1,382 @@
+
+                    UNSAFE.putLong(this.lastParsedNameBytes, Unsafe.ARRAY_BYTE_BASE_OFFSET + totalWordLength, actualWord);
+
+                    totalWordHash ^= actualWord;
+                    totalWordLength += actualLength;
+
+                    this.lastParsedNameLength = totalWordLength;
+                    this.lastParsedNameHash = totalWordHash;
+                    this.currentAddr += totalWordLength + 1; // +1 Because we don't want to point to ';'
+
+                    break;
+                }
+                else {
+                    UNSAFE.putLong(this.lastParsedNameBytes, Unsafe.ARRAY_BYTE_BASE_OFFSET + totalWordLength, actualWord);
+
+                    totalWordLength += 8;
+                    totalWordHash ^= actualWord;
+                }
+            }
+        }
+
+        // Removes "garbage" of a word byte
+        private long mask(long word, int length) {
+            int shift = (8 - length) * 8;
+            return (word << shift) >> shift;
+        }
+
+        private long hasByte(long word, long pattern) {
+            long patternMatch = word ^ pattern;
+            return (patternMatch - 0x0101010101010101L) & (~patternMatch & 0x8080808080808080L);
+        }
+
+        private void adjustStartAddr() {
+            if (currentAddr == this.mmappedFile.address()) {
+                return;
+            }
+
+            while (UNSAFE.getByte(currentAddr) != '\n' && currentAddr != endAddr) {
+                currentAddr++;
+            }
+
+            currentAddr++; // We want it to point to the first character instead of \n
+        }
+
+        private void adjustEndAddr() {
+            long endAddressMmappedFile = mmappedFile.address() + mmappedFileSize;
+            if (endAddr >= endAddressMmappedFile) {
+                return;
+            }
+
+            while (UNSAFE.getByte(endAddr) != '\n' && endAddr != endAddressMmappedFile) {
+                endAddr++;
+            }
+        }
+    }
+
+    static class SimpleMap {
+        private final Result[] entries;
+        private final long size;
+
+        public SimpleMap(int size) {
+            this.entries = new Result[size];
+            this.size = size;
+        }
+
+        public void put(long hashToPut, byte[] nameToPut, int nameLength, int valueToPut) {
+            int index = hashToIndex(hashToPut);
+
+            for (;;) {
+                Result actualEntry = entries[index];
+
+                if (actualEntry == null) {
+                    byte[] nameToPutCopy = new byte[nameLength];
+                    for (int i = 0; i < nameLength; i++) {
+                        nameToPutCopy[i] = nameToPut[i];
+                    }
+
+                    entries[index] = new Result(hashToPut, nameToPutCopy, nameLength, valueToPut,
+                            valueToPut, valueToPut, 1);
+                    return;
+                }
+                if (actualEntry.hash == hashToPut) {
```

Comment:
You'll need to compare the actual name here instead of the hash, as this would still mix up the values from different stations with the same hash.

Reply:
You are right, it is not a correct way of checking for collisions.

 I have committed my solution.